include/oneapi/dpl/pstl/utils.h

```diff
@@ -783,6 +783,26 @@ union __lazy_ctor_storage
     }
 };
 
+// Utility that returns the smallest type in tuple.
+template <typename _Tuple>
+class __min_tuple_type;
+
+template <typename _T>
+class __min_tuple_type<std::tuple<_T>>
+{
+  public:
+    using type = _T;
+};
+
+template <typename _T, typename... _Ts>
+class __min_tuple_type<std::tuple<_T, _Ts...>>
+{
+    using __min_type_ts = typename __min_tuple_type<std::tuple<_Ts...>>::type;
+
+  public:
+    using type = std::conditional_t<(sizeof(_T) < sizeof(__min_type_ts)), _T, __min_type_ts>;
+};
+
```

Comment:
Do you think we will ever want the `type` specifics from a utility like this, or are we always only interested in the `sizeof(type)`?

If its likely to be the latter, we could just make this directly have a public `value` equal to the size of the minimum type rather than a public `type`.

This would simplify the code which uses it as well.

Reply:
We are probably not going to ever use this to just get the type, so I have added this change to instead give a `value` which is the size of the smallest type. Also added a variable template.